src/doc/rustc-dev-guide/src/building/suggested.md

```diff
@@ -59,6 +59,14 @@ always overrides the inner ones.
 
 ## Configuring `rust-analyzer` for `rustc`
 
+### Checking the "library" tree
+
+Checking the "library" tree requires a stage1 compiler, which can be a heavy process on some computers.
+For this reason, bootstrap has a flag called `--skip-std-check-if-no-download-rustc` that skips checking the
+"library" tree if `rust.download-rustc` isn't available. If you want to avoid putting a heavy load on your computer
+with `rust-analyzer`, you can add the `--skip-std-check-if-no-download-rustc` flag to your `./x check` command in
+the `rust-analyzer` configuration.
```

Comment:
There are too many sample RA configurations. I decided to document this flag separately by its own and leave it to users to decide whether or not to utilize it.

Reply:
Ok, fair enough. The main point to get across is that this is only useful when you want to check multiple things (`x check compiler library`). Otherwise if you check only the library you will always require stage 1.

Reply:
Shouldn't the sample file at `src/etc/rust_analyzer_settings.json` be updated?